File: Data Structures & Algorithms/remove-element/submission-4.java
class Solution {
    public int removeElement(int[] nums, int val) {
        // k = 0
        // for i in range(len(nums)):
        //     if nums[i] != val:
        //         nums[k] = nums[i]
        //         k += 1

        // return k

        int k = 0;
        for(int i = 0; i < nums.length; i++){
            if (nums[i] != val){
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sum = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in sum:
                return [sum[diff], i]

            sum[nums[i]] = i

        return []
